pkg/operator/internal/controllers/catapult_controller.go

```diff
@@ -73,6 +73,15 @@ func (r *CatapultReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	if err := r.Get(ctx, req.NamespacedName, catapult); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
+	if catapult.Spec.Paused {
+		if catapult.SetPausedCondition() {
+			if err := r.Client.Status().Update(ctx, catapult); err != nil {
+				return ctrl.Result{}, fmt.Errorf("updating %s status: %w", catapult.Name, err)
+			}
+			// reconciliation paused, skip all other handlers
+			return ctrl.Result{}, nil
+		}
```

Comment:
What if catapult is already paused, and `SetOausedCondition` returns `false`? 
It should also skip reconciliation right?

Reply:
oh, good catch, this return should be in the outer block

Reply:
done